Topics/Tech_Stacks/Axios.md

```diff
@@ -0,0 +1,78 @@
+# Axios: An all-in-one HTTP tool
+
+## What is Axios?
+
+Axios is a JavaScript framework that enables developers to send a variety of HTTP requests, performs automatic and contextual serialization of response data, protects
+against [XSRF](https://en.wikipedia.org/wiki/Cross-site_request_forgery) attacks, and even allows for request cancellation amongst much more. It can be used in both
+the backend and frontend of a full stack application, making it a versatile addition to any JavaScript based application.
+
+## Quick setup and Installation:
+If you are using npm the installation command is simply:
+
+```
+npm install axios
+```
+
+If you are using yarn:
+
+```
+yarn add axios
+```
+
+Once installed, make sure to include the following at the top of any .js file that makes use of axios:
+
+``` JavaScript
+const axios = require('axios');
+```
+
+## Examples:
+Here are a few examples of common use cases of Axios:
+
+### Sending a GET request and working with it's response (Using Arrow functions):
+
+``` JavaScript
+Axios.get("https://www.website.com").then(
+(res) => { console.log(res.data); } // This just logs response data but you would do much more 
+).catch(
+(err) => { console.log(err); } // In the case of an error, we would need to provide some kind of handler
+)
+```
+
+You can also use regular functions as opposed to Arrow functions, as demonstrated in the next section
+
+### Sending a POST request and working with it's response (Using Regular functions):
+
+``` JavaScript
+Axios.post("https://www.website.com", someData).then(
+  function(res){
+    console.log(res.data);
+  }
+).catch(
+  function(err){
+    console.log(err);
+  }
+)
+```
+
+Along with `.get()` and `.post()`, Axios also supports `.put()`, `.patch()`, `.delete()`, and the entirety of the HTTP request frameworks.
```

Comment:
FYI, the result you received is not just the data backend sent, it includes:
where ppl usually need to check res.status first and then use res.data to continue
{
  // `data` is the response that was provided by the server
  data: {},
 
  // `status` is the HTTP status code from the server response
  status: 200,
 
  // `statusText` is the HTTP status message from the server response
  statusText: 'OK',
 
  // `headers` the headers that the server responded with
  // All header names are lower cased
  headers: {},
 
  // `config` is the config that was provided to `axios` for the request
  config: {},
 
  // `request` is the request that generated this response
  // It is the last ClientRequest instance in node.js (in redirects)
  // and an XMLHttpRequest instance the browser
  request: {}
}

Reply:
This is really good information, I've added it in right below the POST request section